src/adapters/livekit.ts

```diff
@@ -0,0 +1,77 @@
+import { IBaseComponent, START_COMPONENT, STOP_COMPONENT } from '@well-known-components/interfaces'
+import { AppComponents } from '../types'
+import { AccessToken } from 'livekit-server-sdk'
+import { Room, RoomEvent } from '@livekit/rtc-node'
+
+export type ILivekitComponent = IBaseComponent
+
+export async function createLivekitComponent(
+  components: Pick<AppComponents, 'config' | 'logs' | 'metrics' | 'dataReceivedHandler'>
+): Promise<ILivekitComponent> {
+  const { config, logs, metrics, dataReceivedHandler } = components
+  const logger = logs.getLogger('livekit')
+
+  const [host, apiKey, apiSecret, roomName, identityPrefix, replicaId] = await Promise.all([
+    config.requireString('LIVEKIT_HOST'),
+    config.requireString('LIVEKIT_API_KEY'),
+    config.requireString('LIVEKIT_API_SECRET'),
+    config.requireString('LIVEKIT_ROOM_NAME'),
+    config.getString('LIVEKIT_IDENTITY_PREFIX'),
+    config.getString('REPLICA_NUMBER')
+  ])
+
+  const identity = `${identityPrefix || 'message-router'}-${replicaId || '0'}`
+
+  const room = new Room()
+
+  room.on(RoomEvent.Connected, () => {
+    logger.info('Connected to LiveKit room')
+    metrics.observe('livekit_connection_status', {}, 1)
+  })
+
+  room.on(RoomEvent.Reconnecting, () => {
+    logger.warn('Reconnecting to LiveKit room')
+    metrics.observe('livekit_connection_status', {}, 0)
+  })
+
+  room.on(RoomEvent.Reconnected, () => {
+    logger.info('Reconnected to LiveKit room')
+    metrics.observe('livekit_connection_status', {}, 1)
+  })
+
+  room.on(RoomEvent.Disconnected, (reason) => {
+    logger.warn('Disconnected from LiveKit room', { reason })
+    metrics.observe('livekit_connection_status', {}, 0)
+  })
+
+  room.on(RoomEvent.DataReceived, dataReceivedHandler.handleMessage(room, identity))
+
+  async function getToken() {
+    const token = new AccessToken(apiKey, apiSecret, {
```

Comment:
A LiveKit access token has a TTL, which if I'm not wrong, it's by default 6hs. I'm not sure what could happen if the server got disconnected after 6hs due to a connection issue, would it be able to connect back to it again or would it need to regenerate the token?